src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryDeployerAutoConfiguration.java

```diff
@@ -192,6 +191,7 @@ public TokenProvider tokenProvider(CloudFoundryConnectionProperties properties)
 			return PasswordGrantTokenProvider.builder()
 				.username(properties.getUsername())
 				.password(properties.getPassword())
+				.loginHint(properties.getLoginHint())
```

Comment:
Given that the `login hint` is an optional property, can we rely on the `PasswordGrantTokenProvider.builder()` to gracefully handle `null` (or empty) login hints? 
Otherwise we will need handle it here. E.g. add a condition to set the login hits only if not empty.

Reply:
yes. setting the hint to null is supported.